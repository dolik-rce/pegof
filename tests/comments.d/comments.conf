input comments.d/comments.peg
