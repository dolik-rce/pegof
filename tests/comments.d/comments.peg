# This is grammar comment

# This is rule comment for 'main'
main <- A / B / C

    # This is indented comment for rule 'A'
A <- "A"

# This is multiline
# comment for rule 'B'
B <- "B"

# This is not really well placed comment for rule 'C'

C <- "C"

# This is comment for code section (which might or might not be present)
%%
