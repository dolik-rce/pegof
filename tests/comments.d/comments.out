# This is grammar comment

# This is rule comment for 'main'
main <-
    A
    / B
    / C
    / D

# This is indented comment for rule 'A'
A <- "A"

# This is multiline
# comment for rule 'B'
B <- "B"

# This is not really well placed comment for rule 'C'
C <- "C"

D <-
 # This is a comment inside rule
 "d"
 / "D" # This is a comment at the end of rule
 / ( "dD" # This is a comment in subsequence
    / "Dd" )

# This is comment for code section (which might or might not be present)
